Display resource load errors seperately from list of items

Nokel81 opened the pull request:
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #4192 
fixes #2683
![Screen Shot 2021-11-02 at 2 57 10 PM](https://user-images.githubusercontent.com/8225332/139928192-7d7f5124-87a3-46e7-9b0f-4b947e425b75.png)

Nokel81 committed:
Display resource load errors seperately from list of items

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jim-docker commented on src/common/k8s-api/kube-watch-api.ts:
```diff
@@ -38,11 +38,38 @@ export interface IKubeWatchEvent<T extends KubeJsonApiData> {
+
+export interface KubeWatchSubscribeStoreOptions extends KubeWatchPreloadOptions {
+  /**
+   * Whether to subsribe only after loading all stores
```
```suggestion
   * Whether to subscribe only after loading all stores
```

jim-docker commented on src/renderer/components/kube-object-list-layout/kube-object-list-layout.tsx:
```diff
@@ -91,6 +101,14 @@ export class KubeObjectListLayout<K extends KubeObject> extends React.Component<
+              <>
+                {info}
+                {this.loadErrors.length > 0 && (
+                  <Icon material="warning" className="load-error" tooltip={this.loadErrors.join("\n")}/>
```
it looks like the newline is not respected when the tooltip is displayed

![Screen Shot 2021-11-10 at 8 05 03 PM](https://user-images.githubusercontent.com/40840436/141218433-7a9d13cc-48ab-4b47-843c-d7b11468cd61.png)

jim-docker reviewed (commented):
I like the UI implementation!

Doesn't always work for All namespaces:

https://user-images.githubusercontent.com/40840436/141219343-8764e32e-fae3-48cb-8637-3b9d72a93f8d.mov

(user has access to only the default namespace)

and a more minor point, the warning icon flashes as you cmd-click to add/remove namespaces from the selection.

Nokel81 committed:
Resolve PR comments

- load errors now are paragraph spaced

- The loading warning stays around when switching to all namespaces

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jim-docker reviewed (commented):
Not sure if this is due to latest commit or was previously there as an edge case:

![Screen Shot 2021-11-15 at 10 50 05 AM](https://user-images.githubusercontent.com/40840436/141812181-cde38273-eddd-4b0d-96e9-8c1fbfc59323.png)

This is seen upon first connection to the cluster, and "All namespaces" is the setting that was loaded from local storage (user can access default namespace only, and can list namespaces. The list should show the pods in the default namespace)

Nokel81 committed:
Move tooltip to bottom, clear store on loadAll cluster scope failure

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jim-docker commented on src/renderer/components/tooltip/tooltip.tsx:
```diff
@@ -112,26 +112,21 @@ export class Tooltip extends React.Component<TooltipProps> {
+    positions.add(TooltipPosition.TOP_RIGHT);
+    positions.add(TooltipPosition.TOP_LEFT);
+    positions.add(TooltipPosition.BOTTOM_RIGHT);
+    positions.add(TooltipPosition.BOTTOM_LEFT);
```
So this kind of refactoring looks like an improvement but makes your work take longer, the review take longer, testing take longer, impacts more code than this PR is intended for, and could introduce new bugs...

Nokel81 replied:
Yes you are right, I shouldn't have committed that, will revert.

Nokel81 committed:
Revert <Tooltip> changes

Signed-off-by: Sebastian Malton <sebastian@malton.name>
